TriblerGUI/widgets/trustpage/js/radial_view/radial_link.js

```diff
@@ -0,0 +1,238 @@
+     */
+    self.selectAll = function () {
+        return svg.select(".links").selectAll(".link");
+    };
+
+    /**
+     * Create, update and destroy the <svg.link> objects based on new data
+     * @param {GraphData} newGraphData
+     */
+    self.onNewData = function (newGraphData) {
+        self.graphData = newGraphData;
+
+        var update = self.selectAll()
+            .data(self.graphData.links, function (link) {
+                return link.id;
+            });
+
+        self.destroy(self.selectAll());
+        self.update(update);
+        self.create(update.enter());
+    };
+
+    /**
+     * Create the new <svg.link> objects based on entering data
+     * @param enterSelection
+     */
+    self.create = function (enterSelection) {
+        // Create <svg.link>
+        var links = enterSelection
+            .append("svg")
+            .attr("class", "link")
+            .style("opacity", "0")
+            .style("stroke", "transparent")
+
+            // Apply stroke width
+            .attr("stroke-width", function (link) {
+                return self._calculateStrokeWidth(link);
+            })
+
+            // Bind mouse events
+            .on("mouseover", self.getEventHandlers("mouseover"))
+            .on("mousemove", self.getEventHandlers("mousemove"))
+            .on("mouseout", self.getEventHandlers("mouseout"));
+
+        // Append <line.link-source> to <svg.link>
+        links.append("line")
+            .attr("class", "link-source")
+            .style("stroke", self.config.color);
+
+        // Append <line.link-target> to <svg.link>
+        links.append("line")
+            .attr("class", "link-target")
+            .style("stroke", self.config.color);
+
+        // Fade-in
+        links
+            .transition()
+            .duration(1000)
+            .style("opacity", function (link) {
+                return self._calculateOpacity(link);
+            });
+
+    };
+
+    /**
+     * Update existing <svg.link> objects based on updated data
+     * @param updateSelection
+     */
+    self.update = function (updateSelection) {
+        updateSelection
+            .attr("stroke-width", function (d) {
+                return self._calculateStrokeWidth(d);
+            });
+    };
+
+    /**
+     * Destroy existing <svg.link> objects based on exiting data
+     * @param exitSelection
+     */
+    self.destroy = function (exitSelection) {
+        exitSelection.remove();
+    };
+
+    /**
+     * Update the positions of the links on each tick.
+     */
+    self.tick = function () {
+        // Part of line at the source
+        self.selectAll().selectAll(".link-source")
+            .attr("x1", function (d) { return d.source.x; })
+            .attr("y1", function (d) { return d.source.y; })
+            .attr("x2", function (d) { return self._xAtFraction(d.source.x, d.target.x, 1 - d.ratio - 1 / Math.max(1, self._nodeDistance(d.source, d.target))); })
```

Comment:
Maybe put this long line in a function, the function name stating what happens here. Makes it more readable and less code duplication.